doc/TESTING.md

```diff
@@ -1,22 +1,100 @@
 # Testing
-This chapter should provide the user with all needed information around testing within your project.
+ This document explains project's testing strategy, including why we write tests and what is our goals are in testing and the tooling library used to write the test cases
 
 ## Content
 - [Different kind of tests](#different-kind-of-tests)
-- [Tooling](#tooling)
+- [Why Do We Write Tests](#why-do-we-write-tests)
+- [Our Goal in Testing](#our-goal-in-testing)
+- [TDD (Test-driven development)](#tdd-test-driven-development)
+- [Tooling](#tooling---testing-libraries-installed)
+- [How to install Jest](#how-to-install-jest)
 - [How to write tests](#how-to-write-tests)
 - [How to run tests](#how-to-run-tests)
 
 ## Different kind of tests
-This section should display all the different kinds of tests that you write in your project. Explain what each kind of test is for and why you use them in your project. You can also give an outlook on new tests that should be added in the future of your project. If you have a QA in place, also mention what kind of tests they perform and add contact persons.
+In this project Unit Testing is the only type of testing done in the development process to ensure that a section of an application (known as the "unit") meets its design
 
+Unit Testing - The main objective of unit testing is to isolate written code to test and determine if it works as intended. This testing method is also the first level of software testing, which is performed before other testing methods such as integration testing
 
-## Tooling
-Please explain what kind of tools you use for testing, why you have choosen them and how to work with them.
+In future other testing types like Integration Testing, Performance Testing can be added in the Continuous Integration process.
 
+## Why Do We Write Tests
+We write tests to verify that our software works as intended and to catch bugs and errors before they pushed and make it into production. Testing helps us catch issues early in the development process, which can save us time and resources in the long run. It also helps ensure that our software meets the requirements.
+
+## Our Goal in Testing
+Our goal is to write automated unit tests for each module or function in our codebase to ensure that it works as intended and the changes do not introduce regressions. The minimum code coverage of unit tests should be **70%**. To achieve this goal, we follow a comprehensive testing strategy of TDD (Test-driven development) process.
```

Comment:
Change the section title to "Testing Goal" Overall, writing should be in 3rd person, not 1st person